packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -216,10 +216,25 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
     'aria-labelledby': ariaLabelledby,
     'aria-invalid': ariaInvalid,
   } = props;
-  const valueFormatter = React.useCallback(date => (date ? formatMonthDayYear(date, dateFormatting) : ''), [
-    dateFormatting,
-    formatMonthDayYear,
-  ]);
+
+  const valueFormatter = React.useCallback(
+    date =>
+      date
+        ? formatMonthDayYear(date, {
+            months: dateFormatting.months,
+            shortMonths: dateFormatting.shortMonths,
+            days: dateFormatting.days,
+            shortDays: dateFormatting.shortDays,
+          })
```

Comment:
Is there a reason for this change? I mean a change from `(date ? formatMonthDayYear(date, dateFormatting) : '')`.

Reply:
Heyy, so `dateFormatting` is always re-created on each render, and it's a pretty huge object, so no point in memo :/ 

For this specific bug, when user type in the `input`, Datepicker re-renders, `dateFormatting` is recreated, so: 
`valueFormatter = date ? formatMonthDayYear(date, dateFormatting) : ''` is recreated, 
and the below useEffect is re-run because of `valueFormatter` is new:
```tsx
  React.useEffect(() => {
    setFormattedDate(valueFormatter(selectedDate));
  }, [selectedDate, valueFormatter]);
```
This useEffect will try to set value in `input`, so user cannot change `input` when they type things